fontforgeexe/prefs.c

```diff
@@ -788,196 +784,17 @@ static char *PrefsUI_getFontForgeShareDir(void) {
-	{ "UTF-8", e_utf8 },
-	{ "ISO-10646/UTF-8", e_utf8 },
-	{ "ISO_10646/UTF-8", e_utf8 },
-	{ "UCS2", e_unicode },
-	{ "UCS-2", e_unicode },
-	{ "UCS-2-INTERNAL", e_unicode },
-	{ "ISO-10646", e_unicode },
-	{ "ISO_10646", e_unicode },
-	/* { "eucJP", e_euc }, */
-	/* { "EUC-JP", e_euc }, */
-	/* { "ujis", ??? }, */
-	/* { "EUC-KR", e_euckorean }, */
-	{ NULL, 0 }
-    };
-
-    int i;
-    char buffer[80];
-#if HAVE_ICONV_H
-    static char *last_complaint;
-
-    iconv_t test;
-    free(iconv_local_encoding_name);
-    iconv_local_encoding_name= NULL;
-#endif
-
-    if ( strchr(enc,'@')!=NULL && strlen(enc)<sizeof(buffer)-1 ) {
-	strcpy(buffer,enc);
-	*strchr(buffer,'@') = '\0';
-	enc = buffer;
-    }
-
-    for ( i=0; encs[i].name!=NULL; ++i )
-	if ( strmatch(enc,encs[i].name)==0 )
-return( encs[i].enc );
-
-    if ( subok ) {
-	for ( i=0; encs[i].name!=NULL; ++i )
-	    if ( strstrmatch(enc,encs[i].name)!=NULL )
-return( encs[i].enc );
-
-#if HAVE_ICONV_H
-	/* I only try to use iconv if the encoding doesn't match one I support*/
-	/*  loading iconv unicode data takes a while */
-	test = iconv_open(enc,FindUnicharName());
-	if ( test==(iconv_t) (-1) || test==NULL ) {
-	    if ( last_complaint==NULL || strcmp(last_complaint,enc)!=0 ) {
-		fprintf( stderr, "Neither FontForge nor iconv() supports your encoding (%s) we will pretend\n you asked for latin1 instead.\n", enc );
-		free( last_complaint );
-		last_complaint = copy(enc);
-	    }
-	} else {
-	    if ( last_complaint==NULL || strcmp(last_complaint,enc)!=0 ) {
-		fprintf( stderr, "FontForge does not support your encoding (%s), it will try to use iconv()\n or it will pretend the local encoding is latin1\n", enc );
-		free( last_complaint );
-		last_complaint = copy(enc);
-	    }
-	    iconv_local_encoding_name= copy(enc);
-	    iconv_close(test);
-	}
-#else
-	fprintf( stderr, "FontForge does not support your encoding (%s), it will pretend the local encoding is latin1\n", enc );
-#endif
-
-return( e_iso8859_1 );
-    }
-return( e_unknown );
-}
-
-static int DefaultEncoding(void) {
-    const char *loc;
-    int enc;
+static void DefaultEncoding(void) {
+    const char* charset = NULL;
+    bool is_utf8 = g_get_charset(&charset);
 
-#if HAVE_LANGINFO_H
-    loc = nl_langinfo(CODESET);
-    enc = encmatch(loc,false);
-    if ( enc!=e_unknown )
-return( enc );
-#endif
-    loc = getenv("LC_ALL");
-    if ( loc==NULL ) loc = getenv("LC_CTYPE");
-    /*if ( loc==NULL ) loc = getenv("LC_MESSAGES");*/
-    if ( loc==NULL ) loc = getenv("LANG");
-
-    if ( loc==NULL )
-return( e_iso8859_1 );
-
-    enc = encmatch(loc,false);
-    if ( enc==e_unknown ) {
-	loc = strrchr(loc,'.');
-	if ( loc==NULL )
-return( e_iso8859_1 );
-	enc = encmatch(loc+1,true);
+    if (!SetupUCharMap(FindUnicharName(), charset, is_utf8)) {
+        fprintf(stderr, "Failed to set up unichar<->system local encoding, assuming utf-8 and trying again...\n");
+        if (!SetupUCharMap(FindUnicharName(), "UTF-8", true)) {
```

Comment:
Previously if iconv setup failed it would default to latin1, I now default to utf8, which I think is more appropriate today.

Reply:
I guess this isn't really related, but `default_encoding` in `startui.c` and `startnoui.c` is still built from `FindOrMakeEncoding("ISO8859-1")`. Given the difficulty you had finding non-unicode fonts, shouldn't we change this to BMP as well?

Reply:
If everyone's fine with changing the default encoding, I'm happy to change it, but maybe as a separate PR. ISO8859-1 is hardcoded in quite a few places.

Reply:
sounds reasonable